lib/Sema/CSSimplify.cpp

```diff
@@ -2090,8 +2090,7 @@ ConstraintSystem::matchTypes(Type type1, Type type2, ConstraintKind kind,
     }
     
     // Special implicit nominal conversions.
-    if (!type1->is<LValueType>() &&
```

Comment:
Making these Subtype rather than Conversion is likely to have other consequences

Reply:
I've talked to @DougGregor offline about this, upcasts were intended for everything `>= Subtype` which is one right before `Conversion` :)